lib/collision_detection.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:math' as math;
+
+import 'extensions/vector2.dart';
+
+/// Checks whether the [polygon] represented by the list of [Vector2] contains
+/// the [point].
+bool containsPoint(Vector2 point, List<Vector2> polygon) {
```

Comment:
Should this file be direct on `lib` or should we make an `util` or something like that folder?

Reply:
I think a lot of the things that live directly `lib` could be in a `util` folder, but I also think that it is unnecessary to introduce a `util` folder before it becomes too many things in the root directory. What are your thoughts?

Reply:
Either way is fine by me, this just hit me because this file "loose" on the `lib` folder seemed a little weird to me, and an `util` folder seemed to make sense. You said that there are other files that you judge worth of being on an `util` folder as well, maybe we could do that as a follow up, or even better, do that on the PR that @renancaraujo has just recently pushed. What do you think?

Reply:
Not completely convinced, I think a `util` folder in general usually gives the hint of that the project isn't well structured, that it just is place where things that doesn't fit anywhere else gets dumped. Which to be fair would be the root now...